File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        
        ArrayList<Integer> p = new ArrayList<>();
        for (int priority : priorities) {
            p.add(priority); 
        }
        
        int answer = 0;
        while(p.size() != 0) {
            int m = Collections.max(p);
            if(m == p.get(location) && location == 0) {
                answer += 1;
                break;
            } else if(m == p.get(0)) {
                answer += 1;
                p.remove(0);
                location -= 1;
            } else if(m != p.get(0)) {
                if (location == 0) {
                    location = p.size() - 1;
                } else {
                    location -= 1;
                }
                p.add(p.remove(0));
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    
    while len(priorities) != 0:
        if max(priorities) == priorities[0] and location == 0:
            answer += 1
            break
        elif priorities[0] != max(priorities):
            if location == 0:
                location = len(priorities) - 1
            else:
                location -= 1
            priorities.append(priorities.pop(0))
        elif priorities[0] == max(priorities):
            answer += 1
            priorities.pop(0)
            location -= 1
            
    return answer